core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrCylinder.java

```diff
@@ -133,35 +133,42 @@ public Point3 getIntersectionInSource(Point3 rv, Ray ray, AffineMatrix4x4 m, int
 
     Point3 cylinderPosition = new Point3(0, 0, 0);
     Vector3 cylinderDirection = new Vector3(0, 0, 0);
-
     Point3 cylinderTopPosition = new Point3(0, 0, 0);
-    if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.POSITIVE_X) {
-      cylinderDirection.x = 1;
-      cylinderPosition.x = bottomValue;
-      cylinderTopPosition.x = topValue;
-    } else if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.POSITIVE_Y) {
-      cylinderDirection.y = 1;
-      cylinderPosition.y = bottomValue;
-      cylinderTopPosition.y = topValue;
-    } else if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.POSITIVE_Z) {
-      cylinderDirection.z = 1;
-      cylinderPosition.z = bottomValue;
-      cylinderTopPosition.z = topValue;
-    } else if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.NEGATIVE_X) {
-      cylinderDirection.x = -1;
-      cylinderPosition.x = -bottomValue;
-      cylinderTopPosition.x = -topValue;
-    } else if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.NEGATIVE_Y) {
-      cylinderDirection.y = -1;
-      cylinderPosition.y = -bottomValue;
-      cylinderTopPosition.y = -topValue;
-    } else if (this.bottomToTopAxis == Cylinder.BottomToTopAxis.NEGATIVE_Z) {
-      cylinderDirection.z = -1;
-      cylinderPosition.z = -bottomValue;
-      cylinderTopPosition.z = -topValue;
-    } else {
-      //todo?
-      throw new RuntimeException();
+
+    switch (bottomToTopAxis) {
+      case POSITIVE_X -> {
+        cylinderPosition = new Point3(bottomValue, 0, 0);
+        cylinderDirection = new Vector3(1, 0, 0);
+        cylinderTopPosition = new Point3(topValue, 0, 0);
+      }
+      case POSITIVE_Y -> {
+        cylinderPosition = new Point3(0, bottomValue, 0);
+        cylinderDirection = new Vector3(0, 1, 0);
+        cylinderTopPosition = new Point3(0, topValue, 0);
+      }
+      case POSITIVE_Z -> {
+        cylinderPosition = new Point3(0, 0, bottomValue);
+        cylinderDirection = new Vector3(0, 0, 1);
+        cylinderTopPosition = new Point3(0, 0, topValue);
+      }
+      case NEGATIVE_X -> {
+        cylinderPosition = new Point3(-bottomValue, 0, 0);
+        cylinderDirection = new Vector3(-1, 0, 0);
+        cylinderTopPosition = new Point3(-topValue, 0, 0);
+      }
+      case NEGATIVE_Y -> {
+        cylinderPosition = new Point3(0, -bottomValue, 0);
+        cylinderDirection = new Vector3(0, -1, 0);
+        cylinderTopPosition = new Point3(0, -topValue, 0);
+      }
+      case NEGATIVE_Z -> {
+        cylinderPosition = new Point3(0, 0, -bottomValue);
+        cylinderDirection = new Vector3(0, 0, -1);
+        cylinderTopPosition = new Point3(0, 0, -topValue);
+      }
+      default ->
+        //todo?
```

Comment:
weird spacing

Reply:
Not sure what you mean.

Reply:
nm, github just formatted it weird, it looks fine in the real file